Improved USER_GUIDE (#27)
rootstrap/react-native-use-styles: USER_GUIDE.md

## User Guide

> Hi, 😄 welcome to the User's guide to using react-native-use-styles. With react-native-use-styles you can easily de-clutter 💆‍♂️ your style code and write a better and more efficient styling code🤩.

Reviewer: ```suggestion
Hi, 😄 welcome to the User's guide to using react-native-use-styles. With `react-native-use-styles` you can easily de-clutter 💆‍♂️ your style code and write a better and more efficient styling code 🤩.
```

> Hi, 😄 welcome to the User's guide to using react-native-use-styles. With `react-native-use-styles` you can easily de-clutter 💆‍♂️ your style code and write a better and more efficient styling code 🤩.

Reviewer: Sorry I didn't saw this but we should use useStyles instead of `react-native-use-styles`:

```suggestion
Hi, 😄 welcome to the User's guide to using useStyles. With useStyles you can easily de-clutter 💆‍♂️ your style code and write a better and more efficient styling code 🤩.
```

This is a comprehensive guide that will let you get started easily.

### Installation

To install the library in your workspace, use the following code.

```
npm install react-native-use-styles --save
```

### Path notation and aliases

Apart from traditional path notations and aliases you can, by using the library, adopt a different approach.

> In the following code we have easily created an aliases 'purple' that will set the background color to purple.

Reviewer: ```suggestion
In the following code we have easily created an aliases `purple` that will set the background color to purple.
```

```js

> import {Styles} from "react-native-use-styles";

Reviewer: ```suggestion
import { Styles } from "react-native-use-styles";
```

export default Styles({

>   purple: 'bg:color:purple',

Reviewer: ```suggestion
  purple: 'bg:color:purple'
```

>   purple: "bg:color:purple",

Reviewer: ```suggestion
  purple: 'bg:color:purple',
```
Can you use single quotes?

});
```

When we use aliases or path notation, the library transforms these paths into valid styles object.

The above given code is equivalent to the approaches given below.

Approach - 1

```js
import { Styles } from 'react-native-use-styles';

export default Styles({

>   purple: 'background:color:purple',

Reviewer: ```suggestion
  purple: 'background:color:purple'
```

> import {Styles} from "react-native-use-styles";

Reviewer: ```suggestion
import { Styles } from "react-native-use-styles";
```

>   purple: "background:color:purple",

Reviewer: ```suggestion
  purple: 'background:color:purple',
```

});
```

Approach - 2

```js
import { Styles } from 'react-native-use-styles';

export default Styles({
  purple: {

>     backgroundColor: 'purple',

Reviewer: ```suggestion
    backgroundColor: 'purple'
```

> import {Styles} from "react-native-use-styles";

Reviewer: ```suggestion
import { Styles } from 'react-native-use-styles';
```

>     backgroundColor: "purple",

Reviewer: ```suggestion
    backgroundColor: 'purple',
```

And a simple styles definition as following:

```js

> import {Styles} from "react-native-use-styles";

Reviewer: ```suggestion
import { Styles } from "react-native-use-styles";
```

export default Styles({

>   purple: {color: "purple"},

Reviewer: ```suggestion
  purple: { color: 'purple' },
```

});
```

### Styles namespace name

Here's the approach for using namespace name.

```js

> import {Styles} from "react-native-use-styles";

Reviewer: ```suggestion
import { Styles } from "react-native-use-styles";
```

> const namespace = "sample";

Reviewer: ```suggestion
const namespace = 'sample';
```

export default Styles(
  {

>     purple: "color:purple",

Reviewer: ```suggestion
    purple: 'color:purple',
```

import { Styles } from 'react-native-use-styles';

export default Styles({

>   purple: { color: 'purple' },

Reviewer: ```suggestion
  purple: { color: 'purple' }
```

});
```

### Styles namespace name

Here's the approach for using namespace name.

```js
import { Styles } from 'react-native-use-styles';

const namespace = 'sample';

export default Styles(
  {

>     purple: 'color:purple',

Reviewer: ```suggestion
    purple: 'color:purple'
```

```

```js

> import {Styles} from "react-native-use-styles";

Reviewer: ```suggestion
import { Styles } from 'react-native-use-styles';
```

export default Styles({

>   anotherNamespace: "fx:1 @sample.purple",

Reviewer: ```suggestion
  anotherNamespace: 'fx:1 @sample.purple',
```

import { Styles } from 'react-native-use-styles';

export default Styles({

>   anotherNamespace: 'fx:1 @sample.purple',

Reviewer: ```suggestion
  anotherNamespace: 'fx:1 @sample.purple'
```

}
```

This cache will be cleared once you unmount the component. The second cache layer is at global level and it's actually where all the GlobalStyles and Styles definition resides. There's a third cache layer, the native layer, we are used to using this cache layer directly in `react-native` when we use the `StyleSheet` API (actually this API seems not to be caching in recent versions). This API caches the styles but in the native thread so you are not sending your styles objects through the bridge every time you use them.

### setSeparator

```js

> import {setSeparator, GlobalStyles} from "react-native-use-styles";

Reviewer: ```suggestion
import { setSeparator, GlobalStyles } from 'react-native-use-styles';
```

> setSeparator("-");

Reviewer: ```suggestion
setSeparator('-');
```

export default GlobalStyles({

>   purple: 'fx-1 fx-direction-row',

Reviewer: ```suggestion
  purple: 'fx-1 fx-direction-row'
```

>   purple: "fx-1 fx-direction-row",

Reviewer: ```suggestion
  purple: 'fx-1 fx-direction-row',
```
